kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/KyuubiRestClient.java

```diff
@@ -36,12 +36,33 @@ public class KyuubiRestClient implements AutoCloseable {
 
   private static final Logger LOG = LoggerFactory.getLogger(KyuubiRestClient.class);
 
+  /** The thread local auth header, used to support multi-tenancy for platform. */
+  private static ThreadLocal<String> threadLocalAuthHeader =
+      new ThreadLocal<String>() {
+        @Override
+        protected synchronized String initialValue() {
+          return null;
+        }
+      };
```

Comment:
cc @ulysses-you  @lightning-L

Reply:
it seems not friendly for user..

We can pull out `getAuthHeader`,`authSchema` and related code from `KyuubiRestClient` to a new interface like `AuthHeaderGenerator`, and pass it to `KyuubiRestClient`.

The interface of `AuthHeaderGenerator` can hold one main methods:
```
generateToken: Array[byte]
```
So developers can implement their multi-user auth header in a custom `AuthHeaderGenerator`. And by default, we can provide two implementation for noauth and SPNEGO.